java-security/src/main/java/com/sap/cloud/security/adapter/xs/XSUserInfoAdapter.java

```diff
@@ -34,40 +48,52 @@
 	static final String ZDN = "zdn";
 	static final String SYSTEM = "SYSTEM";
 	static final String HDB = "HDB";
+
 	private static final Logger LOGGER = LoggerFactory.getLogger(XSUserInfoAdapter.class);
-	private final AccessToken accessToken;
-	private OAuth2ServiceConfiguration configuration;
 
-	public XSUserInfoAdapter(Object accessToken) throws XSUserInfoException {
-		this(accessToken, null);
-	}
+	private final AccessToken accessToken;
+	private final OAuth2ServiceConfiguration configuration;
+	private final OAuth2TokenService oAuth2TokenService;
 
-	public XSUserInfoAdapter(Token accessToken) throws XSUserInfoException {
-		this(accessToken, Environments.getCurrent().getXsuaaConfiguration());
+	public XSUserInfoAdapter(Object accessToken) {
+		this(accessToken, Environments.getCurrent().getXsuaaConfiguration(), new DefaultOAuth2TokenService());
 	}
 
-	public XSUserInfoAdapter(AccessToken accessToken) throws XSUserInfoException {
-		this(accessToken, null);
+	public XSUserInfoAdapter(AccessToken accessToken) {
+		this(accessToken, Environments.getCurrent().getXsuaaConfiguration(), new DefaultOAuth2TokenService());
 	}
 
-	XSUserInfoAdapter(Object accessToken, OAuth2ServiceConfiguration configuration) throws XSUserInfoException {
+	XSUserInfoAdapter(Object accessToken, OAuth2ServiceConfiguration configuration,
+			OAuth2TokenService oAuth2TokenService) {
 		if (!(accessToken instanceof AccessToken)) {
 			String type = Objects.isNull(accessToken) ? null : accessToken.getClass().getName();
 			throw new XSUserInfoException("token is of instance " + type
 					+ " but needs to be an instance of AccessToken.");
 		}
 		this.accessToken = (AccessToken) accessToken;
 		this.configuration = configuration;
+		this.oAuth2TokenService = oAuth2TokenService;
```

Comment:
Is it possible to check, if apache rest client is on class path => create DefaultOAuth2TokenService() or create SpringOAuth2TokenService()